service/arm_templates/main.json

```diff
@@ -0,0 +1,152 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterID": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique ID of this guest cluster."
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_ZRS",
+        "Premium_LRS"
+      ],
+      "metadata": {
+        "description": "Storage Account type."
+      }
+    },
+    "virtualNetworkCidr": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/16",
+      "metadata": {
+        "description": "The main CIDR block reserved for this virtual network."
+      }
+    },
+    "masterSubnetCidr": {
+      "type": "string",
+      "defaultValue": "10.0.1.0/24",
+      "metadata": {
+        "description": "The masters subnet's CIDR."
+      }
+    },
+    "workerSubnetCidr": {
+      "type": "string",
+      "defaultValue": "10.0.2.0/24",
+      "metadata": {
+        "description": "The masters subnet's CIDR."
+      }
+    },
+    "masterLoadBalancerCidr": {
```

Comment:
For AWS we have separate ELBs for `etcd` and `apiserver`. As etcd is internal but apiserver is also used by users. Since the Azure approach is to have separate subnets I think we should split this.

Long term we want to remove the etcd loadbalancer. It's only needed for Calico and they are working on using the k8s API for storage.

Reply:
What I could do is split the masterLoadBalancerCidr into 2 load balancers, masterAPILoadBalancerCidr(public ip) and masterEtcdLoadBalancerCidr (private ip). Does that sound good to you?